modules/core/include/opencv2/core/mat.hpp

```diff
@@ -2364,31 +2364,42 @@ typedef Mat_<Vec3d> Mat3d;
 typedef Mat_<Vec4d> Mat4d;
 
 /** @todo document */
-class CV_EXPORTS UMat
+class CV_EXPORTS_W UMat
 {
 public:
     //! default constructor
-    UMat(UMatUsageFlags usageFlags = USAGE_DEFAULT);
+    CV_WRAP UMat(UMatUsageFlags usageFlags = USAGE_DEFAULT);
     //! constructs 2D matrix of the specified size and type
     // (_type is CV_8UC1, CV_64FC3, CV_32SC(12) etc.)
-    UMat(int rows, int cols, int type, UMatUsageFlags usageFlags = USAGE_DEFAULT);
-    UMat(Size size, int type, UMatUsageFlags usageFlags = USAGE_DEFAULT);
+    CV_WRAP UMat(int rows, int cols, int type, UMatUsageFlags usageFlags = USAGE_DEFAULT);
+    CV_WRAP UMat(Size size, int type, UMatUsageFlags usageFlags = USAGE_DEFAULT);
     //! constucts 2D matrix and fills it with the specified value _s.
-    UMat(int rows, int cols, int type, const Scalar& s, UMatUsageFlags usageFlags = USAGE_DEFAULT);
-    UMat(Size size, int type, const Scalar& s, UMatUsageFlags usageFlags = USAGE_DEFAULT);
+    CV_WRAP UMat(int rows, int cols, int type, const Scalar& s, UMatUsageFlags usageFlags = USAGE_DEFAULT);
+    CV_WRAP UMat(Size size, int type, const Scalar& s, UMatUsageFlags usageFlags = USAGE_DEFAULT);
 
     //! constructs n-dimensional matrix
-    UMat(int ndims, const int* sizes, int type, UMatUsageFlags usageFlags = USAGE_DEFAULT);
-    UMat(int ndims, const int* sizes, int type, const Scalar& s, UMatUsageFlags usageFlags = USAGE_DEFAULT);
+    CV_WRAP UMat(int ndims, const int* sizes, int type, UMatUsageFlags usageFlags = USAGE_DEFAULT);
+    CV_WRAP UMat(int ndims, const int* sizes, int type, const Scalar& s, UMatUsageFlags usageFlags = USAGE_DEFAULT);
+
+#ifdef _BINDINGS_ONLY_CODE
+    //! wrapper-only constructor from a Mat
+    CV_WRAP_CALL(self->v.reset(new cv::UMat(m.getUMat(accessFlags, usageFlags)))) UMat(const Mat& m, int accessFlags = ACCESS_RW, UMatUsageFlags usageFlags = USAGE_DEFAULT);
+
+    //! returns the OpenCL queue used by OpenCV UMat
+    CV_WRAP_CALL(retval = cv::ocl::Queue::getDefault().ptr()) static void* queue() const;
+
+    //! returns the OpenCL context used by OpenCV UMat
+    CV_WRAP_CALL(retval = cv::ocl::Context::getDefault().ptr()) static void* context() const;
+#endif
```

Comment:
This code does not change the interface, and provides the required wrapper-only functions that were introduced by the original wrapper. I hope you would not mind such idea, would you?